xunit.runner.visualstudio.testadapter/Utility/RunSettingsHelper.cs

```diff
@@ -0,0 +1,124 @@
+﻿using Microsoft.VisualStudio.TestPlatform.ObjectModel.Adapter;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+
+#if !PLATFORM_DOTNET
+    using System.Xml;
+#endif
+
+namespace Xunit.Runner.VisualStudio.TestAdapter
+{
+    internal static class RunSettingsHelper
+    {
+        public static bool DisableAppDomain { get; set; }
+
+        public static bool DisableParallelization { get; set; }
+
+        /// <summary>
+        /// Reads settings for the current run from run context
+        /// </summary>
+        /// <param name="runContext">RunContext of the run</param>
+        public static void ReadRunSettings(IRunContext runContext)
+        {
+            var settingsXml = runContext?.RunSettings?.SettingsXml;
+
+#if !PLATFORM_DOTNET
+            if (!string.IsNullOrEmpty(settingsXml))
+            {
+                try
+                {
+                    using (var stringReader = new StringReader(settingsXml))
+                    {
+                        using (var xmlReader = XmlReader.Create(stringReader, new XmlReaderSettings { IgnoreComments = true, IgnoreWhitespace = true }))
+                        {
+                            if (ReadToRunConfigurationElement(xmlReader))
+                            {
+                                // Process the fields in Xml elements
+                                xmlReader.Read();
+                                var numSettingsRead = 0;
+                                while (xmlReader.NodeType == XmlNodeType.Element && numSettingsRead < 2)
+                                {
+                                    string elementName = xmlReader.Name;
```

Comment:
This code got picked up from a legacy code base of ours. Didn't realise that it is still using old XML APIs. Will update.